src/providers/microsoft/azure/mod.rs

```diff
@@ -351,6 +291,50 @@ impl Azure {
         Ok(vmsize)
     }
 
+    /// Fetch SSH public keys from Azure Instance Metadata Service (IMDS)
+    fn fetch_ssh_keys(&self) -> Result<Vec<PublicKey>> {
+        const URL: &str = "metadata/instance/compute/publicKeys?api-version=2021-02-01";
```

Comment:
I think it might be good to add to the above comment linking to the doc's for the url referenced here. Just in case it changes or something.

Reply:
Done!